core/src/main/java/org/infinispan/metrics/impl/MetricsComponentFactory.java

```diff
@@ -74,19 +74,26 @@ private static void logMissingDependencies(Throwable t) {
       log.debug("Micrometer metrics are not available because classpath dependencies are missing.", t);
    }
 
-   private synchronized MetricsRegistry createMetricRegistry() {
+   private synchronized MetricsRegistry createMetricRegistry(ClassLoader classLoader) {
       if (registry != null) {
          return registry;
       }
       if (isMetricsDisabled()) {
          registry = NoMetricRegistry.NO_OP_INSTANCE;
          return registry;
       }
+
       try {
-         registry = new MetricsRegistryImpl();
-      } catch (Throwable t) {
-         logMissingDependencies(t);
-         registry = NoMetricRegistry.NO_OP_INSTANCE;
+         classLoader.loadClass("io.micrometer.prometheusmetrics.PrometheusMeterRegistry");
+         registry = new PrometheusRegistry();
+      } catch (ClassNotFoundException ignore) {
+         try {
+            classLoader.loadClass("io.micrometer.prometheus.PrometheusMeterRegistry");
+            registry = new PrometheusSimpleClientRegistry();
+         } catch (ClassNotFoundException e) {
+            logMissingDependencies(e);
+            registry = NoMetricRegistry.NO_OP_INSTANCE;
```

Comment:
Not related to this original issue, but we can have a second line of defense here:
```suggestion
            MicrometerMeterRegistryConfiguration configuration = globalConfig.module(MicrometerMeterRegistryConfiguration.class);
            MeterRegistry configuredRegistry = configuration == null ? null : configuration.meterRegistry();
            if (configuredRegistry != null) {
               registry = new AbstractMetricsRegistry() {
                  @Override
                  ScrapeRegistry createScrapeRegistry(MeterRegistry registry) {
                     return new NoScrapeRegistry(registry, true);
                  }
               };
               return registry;
            }
            logMissingDependencies(e);
            registry = NoMetricRegistry.NO_OP_INSTANCE;
```

Reply:
Thanks. I have adopted this and updated the debug logging to try to be more informative.